docs/api/lib/error.js

```diff
@@ -0,0 +1,154 @@
+const _ = require('lodash')
+
+const error = require('error')
+
+const getValidationErrorSchemaByKey = validationKey => ({
+  type: 'object',
+  properties: {
+    error: {
+      type: 'string',
+    },
+    code: {
+      type: 'integer',
+      description: 400,
+    },
+    errorv: {
+      type: 'object',
+      properties: {
+        [validationKey]: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              message: {
+                type: 'string',
+              },
+              path: {
+                type: 'array',
+                items: {
+                  type: 'string',
+                },
+              },
+              type: 'string',
+              context: {
+                type: 'object',
+                properties: {
+                  child: {
+                    type: 'string',
+                  },
+                  label: {
+                    type: 'string',
+                  },
+                  value: {
+                    type: 'string',
+                  },
+                  key: {
+                    type: 'string',
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+})
+
+function formatErrorResponse (errorResponse) {
+  // if there is no code then this is a validation error
+  if (_.get(errorResponse, ['properties', 'code'])) {
+    return errorResponse
+  }
+
+  return {
+    type: 'object',
+    properties: {
+      error: {
+        type: 'string',
+        description: errorResponse.error,
+      },
+      code: {
+        type: 'integer',
+        description: errorResponse.code,
+      },
+    },
+  }
+}
+
+function formatErrorResponseByStatus (errorResponsesForStatus) {
+  const formatedErrorResponse = _.map(errorResponsesForStatus, formatErrorResponse)
+
+  const schema = _.size(formatedErrorResponse) > 1 ? {
+    oneOf: formatedErrorResponse,
+  } : formatedErrorResponse[0]
```

Comment:
Not sure if we have that already in use, but I find more readable formatted in this way:

```suggestion
  const schema = _.size(formatedErrorResponse) > 1
    ? {oneOf: formatedErrorResponse}
    : formatedErrorResponse[0]
```